code/modules/power/cell.dm

```diff
@@ -118,9 +118,7 @@
 		rigged = 1 //broken batterys are dangerous
 
 /obj/item/weapon/cell/emp_act(severity)
-	charge -= 1000 / severity
-	if (charge < 0)
-		charge = 0
+	charge = max(charge - maxcharge/severity, 0)
```

Comment:
The big impact of this is of course that, if you have a maxed out hyper cell and get hit from a direct hit of an ion rifle, it should zero out your battery. So this is a big buff to ion rifles, borgs really finally have a reason to fear them. Knocking out APCs in one hit too, so potentially big affects on malf as well. Since the 2sev area goes through windows, you only have to thermite into the core and shoot one of the windows twice.

If my understanding of ion rifle severities is correct, of course.